packages/common/src/hooks/chats/useSetInboxPermissions.ts

```diff
@@ -1,135 +1,86 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useCallback } from 'react'
 
 import type { AudiusSdk } from '@audius/sdk'
-import { ChatPermission } from '@audius/sdk'
 import { useDispatch, useSelector } from 'react-redux'
 
 import { useAppContext } from '~/context/appContext'
 import { Name } from '~/models/Analytics'
-import { Status } from '~/models/Status'
 import { accountSelectors } from '~/store/account'
-import { chatActions, chatSelectors } from '~/store/pages'
+import {
+  chatActions,
+  chatSelectors,
+  InboxSettingsFormValues
+} from '~/store/pages'
+import { transformMapToPermitList } from '~/utils/chatUtils'
 
 const { fetchPermissions } = chatActions
-const { getUserChatPermissions } = chatSelectors
+const { getChatPermissionsStatus, getUserChatPermissions } = chatSelectors
 const { getUserId } = accountSelectors
 
 type useSetInboxPermissionsProps = {
   audiusSdk: () => Promise<AudiusSdk>
 }
 
-const INBOX_PERMISSIONS_SPINNER_TIMEOUT = 1000
-
 export const useSetInboxPermissions = ({
   audiusSdk
 }: useSetInboxPermissionsProps) => {
   const dispatch = useDispatch()
+  const permissions = useSelector(getUserChatPermissions)
   const {
     analytics: { track, make }
   } = useAppContext()
-  const permissions = useSelector(getUserChatPermissions)
+  const permissionsStatus = useSelector(getChatPermissionsStatus)
   const userId = useSelector(getUserId)
-  const currentPermission = permissions?.permits
-  const [permissionStatus, setPermissionStatus] = useState<Status>(Status.IDLE)
-  const [showSpinner, setShowSpinner] = useState(false)
-  const [localPermission, setLocalPermission] = useState<ChatPermission>(
-    currentPermission ?? ChatPermission.ALL
-  )
-
-  // Update local permission state when permissions change in store.
-  // Needed for when user closes/reopens settings before backend has updated.
-  useEffect(() => {
-    setLocalPermission(currentPermission ?? ChatPermission.ALL)
-  }, [currentPermission])
 
   const doFetchPermissions = useCallback(() => {
     if (userId) {
       dispatch(fetchPermissions({ userIds: [userId] }))
     }
   }, [dispatch, userId])
 
-  const _savePermissions = useCallback(
-    async (permission: ChatPermission) => {
-      if (permissionStatus !== Status.LOADING) {
-        setPermissionStatus(Status.LOADING)
-        setShowSpinner(false)
-        // Only show the spinner if saving takes a while
-        setTimeout(
-          () => setShowSpinner(true),
-          INBOX_PERMISSIONS_SPINNER_TIMEOUT
+  const savePermissions = useCallback(
+    async (permitMap: InboxSettingsFormValues) => {
+      let permitList
+      try {
+        const sdk = await audiusSdk()
+        permitList = transformMapToPermitList(permitMap)
+        await sdk.chats.permit({ permitList, allow: true })
```

Comment:
how do you unset permissions if `allow: true`?

Reply:
`allow` is just the name of the flag we're using to differentiate between the old/new API. the existence of the permit like 'all', 'tippers' indicates backend should set that permit to true, and if it's not in the list, should set it to false.